### PR TITLE
Adding the ability to specify a path using a regular expression

### PR DESCRIPTION
**Pull request by slukes**
We have a usecase where we need to match a url containing a query parameter which we do not know the exact value of in advance.  The option to use a regex for the path enables to acheive this neatly and with out changing production code to put some fixed value in the test which is the current solution.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -152,8 +162,18 @@ For using REST-assured from non-ruby environments.
     $ curl http://localhost:4578/api/something
     awesome
 ```
+  Example using a regular expression for the path:
+  ```
+      $ curl -d 'fullpath=/api/something&content=awesome&response_headers%5BContent-Type%5D=text%2Fhtml' http://localhost:4578/doubles
```
Comment (artemave) [inline README.markdown]: This is still using `fullpath`. Isn't it supposed to be `pathpattern`?

File: README.markdown [hunk 2]
```diff
@@ -89,7 +89,17 @@ Create double:
...
+
+```ruby
+RestAssured::Double.create(pathpattern: /products\/[a-z]{10}\/?param=.*/, content: 'this is more content')
+```
+
+Now GET `http://localhost:4578/products` will be returning `this is content` and a GET to
+`http://localhost:4578/products/coolprod22?param=foo` will be returning `this is more content`.
+
+The pathpattern parameter must be a ruby Regexp object or a string containing a valid regular expression.
+Don't forget you will need to escape thing this slashes and question marks in the pattern.
```
Comment (artemave) [inline README.markdown]: I'd skip this paragraph: `Don't forget you will need to escape thing this slashes and question marks in the pattern.`

This readme is not the right place to teach people how to use regular expressions.

File: features/step_definitions/doubles_steps.rb [hunk 3]
```diff
@@ -82,8 +100,10 @@
 end
 
 Then /^I should see existing doubles:$/ do |doubles|
+  save_and_open_page
```
Comment (artemave) [inline features/step_definitions/doubles_steps.rb]: leftover?

**Comment by slukes** [comment]
@artemave those comments have now been addressed - thanks

**Comment by artemave** [comment]
Pushed new version of gem. Thank you!

**Pull request merged**